tests/Unit/Core/Configuration/MultistoreConfiguratorTest.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use PrestaShop\PrestaShop\Adapter\Configuration as ShopConfiguration;
+use PrestaShop\PrestaShop\Adapter\Shop\Context as ShopContext;
+use PrestaShop\PrestaShop\Core\Configuration\MultistoreConfigurator;
+
+class MultistoreConfiguratorTest extends TestCase
+{
+    protected function setUp()
+    {
+        $this->mockedShopConfiguration = $this->createShopConfigurationMock();
+        parent::setUp();
+    }
+
+    /**
+     * @dataProvider provideForTestRemoveDisabledFields
+     *
+     * @param array $fields
+     * @param bool $isAllShopContext
+     * @param array $expectedResult
+     */
+    public function testRemoveDisabledFields(array $fields, bool $isAllShopContext, array $expectedResult): void
+    {
+        $multistoreConfigurator = $this->getTestableClass($isAllShopContext);
+        $result = $multistoreConfigurator->removeDisabledFields($fields);
+
+        $this->assertEquals($expectedResult, $result);
+    }
+
+    public function provideForTestRemoveDisabledFields()
+    {
+        return [
+            [
+                ['test_field' => 'test_value'],
+                true,
+                ['test_field' => 'test_value'],
+            ],
+            [
+                ['test_field' => 'test_value'],
+                false,
+                ['test_field' => 'test_value'],
+            ],
+            [
+                ['test_field' => 'test_value', 'multistore_test_field' => true],
+                false,
+                ['test_field' => 'test_value', 'multistore_test_field' => true],
+            ],
+            [
+                ['test_field' => 'test_value', 'multistore_test_field' => false],
+                false,
+                ['multistore_test_field' => false],
+            ],
+        ];
```

Comment:
You should add a test with `'multistore_test_field' => false` but `$isAllShopContext = true` To check that it is ignored even if present

Reply:
This situation can't exist if the MultistoreCheckboxEnabler does his job right, but...

In the eventuality that `multistore_test_field` exists in "all shop" context, I'm not even sure what behavior we should have, I'd say we should not remove it, someone might have hardcoded it in the formType for a reason...

So I add this test, checking that the fields are still untouched, even if there is a `multistore_test_field` field.